doc/iso-embed-ignition.md

```diff
@@ -0,0 +1,30 @@
+# ISO-embedded Ignition configuration
+
+CoreOS ISO images are typically used for live-booting machines directly from a read-only storage device (e.g. a CDROM) on non-cloud platforms.
+
+First-boot provisioning via Ignition in such environments is difficult, as there are no reliable metadata endpoints, hypervisor back-channels, or writable disks.
+
+For such reasons `coreos-installer` supports a special mode for ISO images, where an Ignition configuration file can be embedded as a user-customization into a pristine image.
+The resulting image can be then used to boot a live system which is provisioned with the given Ignition configuration.
+
+## Technical details
+
+The ISO-embedding mechanism works by modifying some raw data directly on the image.
+The technical specifcations are described below to help third-party logic.
+
+CoreOS ISO images come with some reserved empty space, the "embed area", which can be used to inject the Ignition configuration.
+The embed area is a block of zero-filled padding concatenated to the end of the initrd file.
+
+The location and size of the embed area depends on each image, and can be read from each disk ISO9660 System Area (i.e. the first 32KiB of the disk).
+The last 24 bytes of this 32 KiB area will contains the following fields, in this order and without special padding/alignment:
+
+ * magic value (8 bytes): the ASCII value "coreiso+" as a marker to identify CoreOS ISO images.
+ * location (64 bits): little-endian unsigned 64-bits integer with the offset for the starting byte of the embed area.
```

Comment:
It'd be more consistent to always use `(8 bytes)` rather than switching to `(64 bits)` for the integers.